qt/python/mantidqt/utils/test/test_modal_tester.py

```diff
@@ -0,0 +1,101 @@
+
+from mock import patch
+import unittest
+
+from qtpy.QtWidgets import QInputDialog
+
+from mantidqt.utils.qt.plugins import setup_library_paths
+from mantidqt.utils.qt.testing import ModalTester
+
+
+setup_library_paths()
+
+
+class TestModalTester(unittest.TestCase):
+
+    def test_pass_widget_closed(self):
+        def testing_function(widget):
+            widget.setTextValue('Hello World!')
+            text = widget.textValue()
+            self.assertEqual(text, 'Hello World!')
+            widget.accept()
+        tester = ModalTester(QInputDialog, testing_function)
+        tester.start()
+        self.assertFalse(tester.widget.isVisible())
+        self.assertTrue(tester.passed)
+
+    def test_fail_widget_closed(self):
+        def testing_function(widget):
+            widget.setTextValue('Hello World!')
+            text = widget.textValue()
+            self.assertEqual(text, 'STUFF')
+            widget.accept()
+        with patch('traceback.print_exc') as print_exc:
+            tester = ModalTester(QInputDialog, testing_function)
+            tester.start()
+            print_exc.assert_called_once_with()
+        self.assertFalse(tester.widget.isVisible())
+        self.assertFalse(tester.passed)
+
+    def test_pass_widget_open(self):
+        def testing_function(widget):
+            widget.setTextValue('Hello World!')
+            text = widget.textValue()
+            self.assertEqual(text, 'Hello World!')
+        tester = ModalTester(QInputDialog, testing_function)
+        tester.start()
+        self.assertFalse(tester.widget.isVisible())
+        self.assertTrue(tester.passed)
+
+    def test_fail_widget_open(self):
+        def testing_function(widget):
+            widget.setTextValue('Hello World!')
+            text = widget.textValue()
+            self.assertEqual(text, '')
+        with patch('traceback.print_exc') as print_exc:
+            tester = ModalTester(QInputDialog, testing_function)
+            tester.start()
+            print_exc.assert_called_once_with()
+        self.assertFalse(tester.widget.isVisible())
+        self.assertFalse(tester.passed)
+
+    def test_exception(self):
+        def testing_function(widget):
+            raise RuntimeError('')
+        with patch('traceback.print_exc') as print_exc:
+            tester = ModalTester(QInputDialog, testing_function)
+            tester.start()
+            print_exc.assert_called_once_with()
+        self.assertFalse(tester.widget.isVisible())
+        self.assertFalse(tester.passed)
+
+    def test_exception_in_creator(self):
+        def create():
+            raise RuntimeError('')
+
+        def testing_function(widget):
+            pass
+        with patch('traceback.print_exc') as print_exc:
+            tester = ModalTester(create, testing_function)
+            tester.start()
+            print_exc.assert_called_once_with()
+        self.assertTrue(tester.widget is None)
+        self.assertFalse(tester.passed)
```

Comment:
I think this file is missing the usual
```
if __name__ == "__main__":
    unittest.main()
```
and has not been added in the CMakeLists.txt?